code-challenges.test.js

```diff
@@ -29,29 +56,116 @@ const colors2 = [
+// remove the last element and put in front
+// utilize the reduce method to further reorder the elements into a new array
+// to even further randomize the elements we call the shuffler2 a random number of times.
+
+
+//im actually pretty proud of how that one turned out
+
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
+
+describe("voteCounter", () => {
+  it("takes in an object that contains up votes and down votes and returns the end tally.", () => {
+    const votes1 = { upVotes: 13, downVotes: 2 }
+    // Expected output: 11
+    const votes2 = { upVotes: 2, downVotes: 33 }
+    // Expected output: -31
+  expect(voteCounter(votes1)).toEqual(11)
+  expect(voteCounter2(votes2)).toEqual(-31)
+})
+})
+
 // a) Create a test with expect statements for each of the variables provided.
 
 const votes1 = { upVotes: 13, downVotes: 2 }
 // Expected output: 11
 const votes2 = { upVotes: 2, downVotes: 33 }
 // Expected output: -31
 
+const voteCounter = (object) =>  object.upVotes - object.downVotes
+
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
+// Pseudo code: 
+// create a function called voteCounter that has an object as its only parameter
+// access the values by using dot notation and subtract them
+// the above method relies on us knowing the names of he keys beforehand.
+// the below method relies on us knowing the order of the keys beforehand.
+
+
+const voteCounter2 = (object) => Object.values(object)[0] - Object.values(object)[1]
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
+describe("newArrayBuilder", () => {
+  it("takes in two arrays as arguments and returns one array with no duplicate values.", () => {
+    const dataTypesArray1 = ["array", "object", "number", "string", "Boolean"]
+    const dataTypesArray2 = ["string", "null", "Boolean", "string", "undefined"]
+    expect(newArrayBuilder(dataTypesArray1, dataTypesArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+    expect(newArrayBuilder2(dataTypesArray1, dataTypesArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+})
+})
+
 // a) Create a test with an expect statement using the variables provided.
 
 const dataTypesArray1 = ["array", "object", "number", "string", "Boolean"]
 const dataTypesArray2 = ["string", "null", "Boolean", "string", "undefined"]
 // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
 
+const newArrayBuilder = (arr1, arr2) => {
+  return arr1.concat(arr2).filter((value, index, arr) => arr.indexOf(value) == index)
+
+}
+const newArrayBuilder2 = (...arr) => {
+  return [].concat(...arr).filter((value, index, arr) => arr.indexOf(value) == index)
```

Comment:
Nice job figuring out the stretch! I knew the spread operator king would figure that out easy. Make sure you are using strict equality though!